spec/workers/notifications/remove_all_worker_spec.rb

```diff
@@ -1,17 +1,16 @@
 require "rails_helper"
 
-RSpec.describe Notifications::RemoveAllJob, type: :job do
-  include_examples "#enqueues_job", "remove_all_notifications", {}
-
-  describe "#perform_now" do
+RSpec.describe Notifications::RemoveAllWorker, type: :woker do
+  describe "#perform" do
+    let(:worker) { subject }
     let(:remove_all_service) { double }
 
     before do
       allow(remove_all_service).to receive(:call)
     end
 
     it "calls the service" do
-      described_class.perform_now(1, "Article", remove_all_service)
+      worker.perform(1, "Article", remove_all_service)
       expect(remove_all_service).to have_received(:call).with([1], "Article")
```

Comment:
@mstruve The problem that I was [talking](https://github.com/thepracticaldev/dev.to/pull/5451#discussion_r366069437) lies here. On my local machine, I removed the third arg from the spec but now the spec is failing with this message:
```shell
1) Notifications::RemoveAllWorker#perform calls the service
     Failure/Error: expect(remove_all_service).to have_received(:call).with([1], "Article")
     
       (Double (anonymous)).call([1], "Article")
           expected: 1 time with arguments: ([1], "Article")
           received: 0 times
     # ./spec/workers/notifications/remove_all_worker_spec.rb:14:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:95:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:95:in `block (2 levels) in <top (required)>'
```

Reply:
Right, that will fail bc we no longer are passing in a stubbed service and therefore that stub( `remove_all_service` ) will not receive the method `call`. You have to rewrite the spec to handle the new worker so you will want something along these lines. 

```ruby
RSpec.describe Notifications::RemoveAllWorker, type: :woker do
  include_examples "#enqueues_on_correct_queue", "low_priority", 1

  describe "#perform_now" do
    let(:worker) { subject }

    it "calls the service" do
      allow(Notifications::RemoveAll).to receive(:call)
      worker.perform(1, "Article")
      expect(Notifications::RemoveAll).to have_received(:call).with([1], "Article")
    end
  end
end
```
Try that and run the spec locally. If it works you are good to go, if it doesn't read the error message and try to decipher what the code is doing. You might need other tweaks here and there. Feel free to jump in with a `binding.pry` or some puts statements if you need to in order to get to the bottom of it. Nothing is off limits when you are trying to get code to work.

Reply:
I think it's because you don't need `remove_all_service` at all anymore. You have to change the expectation to the class you're calling inside the job

Reply:
The code of @mstruve works, thanks. I agree with you @rhymes.